### DISCUSSION TITLE
First experience of localization using C++ and RTAB map using saved map

### DISCUSSION THREAD

**Discussion by damavand1** [introlab/rtabmap#1087] [General]
I installed Ros2 humble on ubuntu 22.04
I installed Rtabmap using this command -> sudo apt install ros-humble-rtabmap-ros
I created 3D map of my room and saved it in db file

now i want to write C++ code that print X,Y,Z,roll,pitch,yaw on screen every 1 second

can you please give me a sample for this ?

**Comment by matlabbe** [comment]
Did you try `ros2 run tf2_ros tf2_echo map base_link 1` ([reference](https://docs.ros.org/en/foxy/Tutorials/Intermediate/Tf2/Introduction-To-Tf2.html#using-tf2-echo)) ?
